### PR TITLE
648 rounding number

### PR DESCRIPTION
**Pull request by mtgoncalves1**
## Overview
Closes https://gitlab.com/architect-io/architect-cli/-/issues/648.  
Fix rounding issue when reading local environment variable with `dotenv`.

## Changes
If the secret value get rounded when casting to a number, use its string format instead.

## Tests
Export an environment variable    
$ export ARC_world_text=1232456789.1232456789   
Run hello-world example    
$ architect dev   

I tested it this way instead of using the `dotenv` command, `dotenv -e .env architect dev`, because the new code doesn't seem to be picked up.

Test remote   
Use package `"@architect-io/cli": "1.37.0-arc-mary.1"` in the API.    
$ architect-local register  
$ architect-dev register  
Deploy it with the following `world_text` value  
Screenshot 2023-04-04 at 10 00 19 AM
Screenshot 2023-04-04 at 10 07 55 AM

### CODE REVIEW

File: src/common/utils/deploy.utils.ts [hunk 1]
```diff
@@ -7,18 +7,16 @@ import yaml from 'js-yaml';
...
 
     for (const [secret_name, secret_value] of Object.entries(process.env || {})) {
       if (secret_name.startsWith('ARC_')) {
         const key = secret_name.substring(4);
-        let value: string | number | undefined = secret_value;
-        if (value && isNumberString(value)) {
-          value = Number.parseFloat(value);
-        }
+        const value = parseValue(secret_value);
         extra_secrets[key] = value;
```
Comment (ryan-cahill) [inline src/common/utils/deploy.utils.ts:20]: `parseValue(secret_value);` can just be set here as `extra_secrets[key] = parseValue(secret_value);`

File: src/common/utils/secret/helper.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { isNumberString } from 'class-validator';
+
+export const parseValue = (secret_value: string | undefined): (string | number | null) => {
```
Comment (ryan-cahill) [inline src/common/utils/secret/helper.ts:3]: This function name could be a little more specific. Maybe something like `parseEnvironmentVariable`?

File: src/common/utils/deploy.utils.ts [hunk 3]
```diff
@@ -7,18 +7,16 @@ import yaml from 'js-yaml';
...
+    const extra_secrets: { [s: string]: string | number | null } = {};
 
     for (const [secret_name, secret_value] of Object.entries(process.env || {})) {
       if (secret_name.startsWith('ARC_')) {
         const key = secret_name.substring(4);
-        let value: string | number | undefined = secret_value;
-        if (value && isNumberString(value)) {
-          value = Number.parseFloat(value);
-        }
+        const value = parseValue(secret_value);
```
Comment (ryan-cahill) [inline src/common/utils/deploy.utils.ts:19]: This function should also be used in the loop below - https://github.com/architect-team/architect-cli/pull/876/files#diff-70a60bd137ea6b541ead0a94f4eba0ffa6a04ee3d84b15b98cdb6d03bf2eb3edR24

This would be necessary if someone uses one or more `-s` flags such as `architect dev -s this=that -s another=1.999999999999999`

File: src/common/utils/secret/helper.ts
Comment (ryan-cahill) [inline src/common/utils/secret/helper.ts:1]: In order to use this in the API from the CLI package, this would need to be exported from `src/index.ts`. It would be good to test with an `arc-` branch to be sure

**Comment by mueschm** [comment]
I guess I am a bit confused as to why not just use the string value all the time then? Like if we have an edge case where we need to treat numbers as strings, it means every downstream system needs to handle that edge case anyway. So even allowing it to be a number here feels odd. 

I know the main reason we allow this stuff is for things like `cpu` on the spec to be interpolated during deploy time from a secret. So the real question is should we just have secrets always be string and have the validation done by each resource on the spec itself?

@tjhiggins what do you think?

**Review by ryan-cahill — CHANGES_REQUESTED** [review]
I know I didn't mark the ticket with the `tests-required` tag at first, but can you add some for the relevant cases please?

File: src/common/utils/secret/helper.ts (hunk 2 above)
Comment (mtgoncalves1) [reply src/common/utils/secret/helper.ts:3]: Yes, `parseEnvironmentVariable` definitely sounds better.

**Review by ryan-cahill — CHANGES_REQUESTED** [review]
This looks good, but can you please include a test for variable rounding when an `ARC_` environment variable is set? You can include one in a test by using something like

```ts
test
  .env({ 'ARC_a_required_key': '123456789.123456789' })
...
```

**Review by ryan-cahill — APPROVED** [review]
